### PR TITLE
[SPARK-30888][CORE][DOC] Add version information to the configuration of Network

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
1.Add version information to the configuration of `Network`.
2.Update the docs of `Network`.

I sorted out some information show below.

Item name | Since version | JIRA ID | Commit ID | Note
-- | -- | -- | -- | --
spark.network.crypto.saslFallback | 2.2.0 | SPARK-19139 | 8f3f73abc1fe62496722476460c174af0250e3fe#diff-0ac65da2bc6b083fb861fe410c7688c2 |  
spark.network.crypto.enabled | 2.2.0 | SPARK-19139 | 8f3f73abc1fe62496722476460c174af0250e3fe#diff-6bdad48cfc34314e89599655442ff210 |  
spark.network.remoteReadNioBufferConversion | 2.4.0 | SPARK-24307 | 2c82745686f4456c4d5c84040a431dcb5b6cb60b#diff-2b643ea78c1add0381754b1f47eec132 |  
spark.network.timeout | 1.3.0 | SPARK-4688 | d3f07fd23cc26a70f44c52e24445974d4885d58a#diff-1df6b5af3d8f9f16255ff8c7a06f402f |  
spark.network.timeoutInterval | 1.3.2 | SPARK-5529 | ec196ab1c7569d7ab0a50c9d7338c2835f2c84d5#diff-47779b72f095f7e7f926898fa1a425ee |  
spark.rpc.askTimeout | 1.4.0 | SPARK-6490 | 8136810dfad12008ac300116df7bc8448740f1ae#diff-529fc5c06b9731c1fbda6f3db60b16aa |  
spark.rpc.connect.threads | 1.6.0 | SPARK-6028 | 084e4e126211d74a79e8dbd2d0e604dd3c650822#diff-0c89b4a60c30a7cd2224bb64d93da942 |  
spark.rpc.io.numConnectionsPerPeer | 1.6.0 | SPARK-10745 | 34a77679877bc40b58a10ec539a8da00fed7db39#diff-0c89b4a60c30a7cd2224bb64d93da942 |  
spark.rpc.io.threads | 1.6.0 | SPARK-6028 | 084e4e126211d74a79e8dbd2d0e604dd3c650822#diff-0c89b4a60c30a7cd2224bb64d93da942 |  
spark.rpc.lookupTimeout | 1.4.0 | SPARK-6490 | 8136810dfad12008ac300116df7bc8448740f1ae#diff-529fc5c06b9731c1fbda6f3db60b16aa |  
spark.rpc.message.maxSize | 2.0.0 | SPARK-7997 | bc1babd63da4ee56e6d371eb24805a5d714e8295#diff-529fc5c06b9731c1fbda6f3db60b16aa |  
spark.rpc.netty.dispatcher.numThreads | 1.6.0 | SPARK-11079 | 1797055dbf1d2fd7714d7c65c8d2efde2f15efc1#diff-05133dfc4bfdb6a27aa092d86ce24866 |  
spark.rpc.numRetries | 1.4.0 | SPARK-6490 | 8136810dfad12008ac300116df7bc8448740f1ae#diff-529fc5c06b9731c1fbda6f3db60b16aa |  
spark.rpc.retry.wait | 1.4.0 | SPARK-6490 | 8136810dfad12008ac300116df7bc8448740f1ae#diff-529fc5c06b9731c1fbda6f3db60b16aa |  

### Why are the changes needed?
Supplemental configuration version information.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Exists UT

### CODE REVIEW

**Comment by maropu** [comment]
For easy-to-cherrypick to branch-3.0, I think its better to split the PR (to add the versions) into too small parts... we really need 10 subtickets for that? https://issues.apache.org/jira/browse/SPARK-30839

**Comment by beliefer** [comment]
> For easy-to-cherrypick to branch-3.0, I think its better to split the PR (to add the versions) into too small parts... we really need 10 subtickets for that? https://issues.apache.org/jira/browse/SPARK-30839

First, the process of sorting is laborious. Secondly, this segmentation is convenient for PR review, otherwise there is too much content in a PR.

**Comment by HyukjinKwon** [comment]
@beliefer, It's trouble some to dig the history so I am okay but I think for some tickets, maybe multiple tickets can be fixed in single PR. That will alleviate @maropu's concern.
BTW, https://github.com/apache/spark/commit/59d6d5cbb04adbd878282b04f0e4b8f7f775c2bd was only merged to master so we won't probably backport to branch-3.0.

**Comment by beliefer** [comment]
@HyukjinKwon Thanks for your understand. I dig the version number according to the file. This will bring some convenience to my work.

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 1]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
 
   private[spark] val NETWORK_CRYPTO_SASL_FALLBACK =
     ConfigBuilder("spark.network.crypto.saslFallback")
+      .version("2.2.0") // SPARK-19139, commit ID: 8f3f73abc1fe62496722476460c174af0250e3fe
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-19139, commit ID: 8f3f73abc1fe62496722476460c174af0250e3fe

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 2]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .booleanConf
       .createWithDefault(true)
 
   private[spark] val NETWORK_CRYPTO_ENABLED =
     ConfigBuilder("spark.network.crypto.enabled")
+      .version("2.2.0") // SPARK-19139, commit ID: 8f3f73abc1fe62496722476460c174af0250e3fe
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-19139, commit ID: 8f3f73abc1fe62496722476460c174af0250e3fe

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 3]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .booleanConf
       .createWithDefault(false)
 
   private[spark] val NETWORK_REMOTE_READ_NIO_BUFFER_CONVERSION =
     ConfigBuilder("spark.network.remoteReadNioBufferConversion")
+      .version("2.4.0") // SPARK-24307, commit ID: 2c82745686f4456c4d5c84040a431dcb5b6cb60b
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-24307, commit ID: 2c82745686f4456c4d5c84040a431dcb5b6cb60b

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 4]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .booleanConf
       .createWithDefault(false)
 
   private[spark] val NETWORK_TIMEOUT =
     ConfigBuilder("spark.network.timeout")
+      .version("1.3.0") // SPARK-4688, commit ID: d3f07fd23cc26a70f44c52e24445974d4885d58a
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-4688, commit ID: d3f07fd23cc26a70f44c52e24445974d4885d58a

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 5]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .timeConf(TimeUnit.SECONDS)
       .createWithDefaultString("120s")
 
   private[spark] val NETWORK_TIMEOUT_INTERVAL =
     ConfigBuilder("spark.network.timeoutInterval")
+      .version("1.3.2") // SPARK-5529, commit ID: ec196ab1c7569d7ab0a50c9d7338c2835f2c84d5
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-5529, commit ID: ec196ab1c7569d7ab0a50c9d7338c2835f2c84d5

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 6]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString(STORAGE_BLOCKMANAGER_TIMEOUTINTERVAL.defaultValueString)
 
   private[spark] val RPC_ASK_TIMEOUT =
     ConfigBuilder("spark.rpc.askTimeout")
+      .version("1.4.0") // SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 7]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .stringConf
       .createOptional
 
   private[spark] val RPC_CONNECT_THREADS =
     ConfigBuilder("spark.rpc.connect.threads")
+      .version("1.6.0") // SPARK-6028, commit ID: 084e4e126211d74a79e8dbd2d0e604dd3c650822
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6028, commit ID: 084e4e126211d74a79e8dbd2d0e604dd3c650822

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 8]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createWithDefault(64)
 
   private[spark] val RPC_IO_NUM_CONNECTIONS_PER_PEER =
     ConfigBuilder("spark.rpc.io.numConnectionsPerPeer")
+      .version("1.6.0") // SPARK-10745, commit ID: 34a77679877bc40b58a10ec539a8da00fed7db39
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-10745, commit ID: 34a77679877bc40b58a10ec539a8da00fed7db39

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 9]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createWithDefault(1)
 
   private[spark] val RPC_IO_THREADS =
     ConfigBuilder("spark.rpc.io.threads")
+      .version("1.6.0") // SPARK-6028, commit ID: 084e4e126211d74a79e8dbd2d0e604dd3c650822
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6028, commit ID: 084e4e126211d74a79e8dbd2d0e604dd3c650822

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 10]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createOptional
 
   private[spark] val RPC_LOOKUP_TIMEOUT =
     ConfigBuilder("spark.rpc.lookupTimeout")
+      .version("1.4.0") // SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 11]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .stringConf
       .createOptional
 
   private[spark] val RPC_MESSAGE_MAX_SIZE =
     ConfigBuilder("spark.rpc.message.maxSize")
+      .version("2.0.0") // SPARK-7997, commit ID: bc1babd63da4ee56e6d371eb24805a5d714e8295
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-7997, commit ID: bc1babd63da4ee56e6d371eb24805a5d714e8295

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 12]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createWithDefault(128)
 
   private[spark] val RPC_NETTY_DISPATCHER_NUM_THREADS =
     ConfigBuilder("spark.rpc.netty.dispatcher.numThreads")
+      .version("1.6.0") // SPARK-11079, commit ID: 1797055dbf1d2fd7714d7c65c8d2efde2f15efc1
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-11079, commit ID: 1797055dbf1d2fd7714d7c65c8d2efde2f15efc1

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 13]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createOptional
 
   private[spark] val RPC_NUM_RETRIES =
     ConfigBuilder("spark.rpc.numRetries")
+      .version("1.4.0") // SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 14]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createWithDefault(3)
 
   private[spark] val RPC_RETRY_WAIT =
     ConfigBuilder("spark.rpc.retry.wait")
+      .version("1.4.0") // SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 15]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
 
   private[spark] val NETWORK_CRYPTO_SASL_FALLBACK =
     ConfigBuilder("spark.network.crypto.saslFallback")
+      .version("2.2.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-19139, commit ID: 8f3f73abc1fe62496722476460c174af0250e3fe

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 16]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .booleanConf
       .createWithDefault(true)
 
   private[spark] val NETWORK_CRYPTO_ENABLED =
     ConfigBuilder("spark.network.crypto.enabled")
+      .version("2.2.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-19139, commit ID: 8f3f73abc1fe62496722476460c174af0250e3fe

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 17]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .booleanConf
       .createWithDefault(false)
 
   private[spark] val NETWORK_REMOTE_READ_NIO_BUFFER_CONVERSION =
     ConfigBuilder("spark.network.remoteReadNioBufferConversion")
+      .version("2.4.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-24307, commit ID: 2c82745686f4456c4d5c84040a431dcb5b6cb60b

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 18]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .booleanConf
       .createWithDefault(false)
 
   private[spark] val NETWORK_TIMEOUT =
     ConfigBuilder("spark.network.timeout")
+      .version("1.3.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-4688, commit ID: d3f07fd23cc26a70f44c52e24445974d4885d58a

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 19]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .timeConf(TimeUnit.SECONDS)
       .createWithDefaultString("120s")
 
   private[spark] val NETWORK_TIMEOUT_INTERVAL =
     ConfigBuilder("spark.network.timeoutInterval")
+      .version("1.3.2")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-5529, commit ID: ec196ab1c7569d7ab0a50c9d7338c2835f2c84d5

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 20]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString(STORAGE_BLOCKMANAGER_TIMEOUTINTERVAL.defaultValueString)
 
   private[spark] val RPC_ASK_TIMEOUT =
     ConfigBuilder("spark.rpc.askTimeout")
+      .version("1.4.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 21]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .stringConf
       .createOptional
 
   private[spark] val RPC_CONNECT_THREADS =
     ConfigBuilder("spark.rpc.connect.threads")
+      .version("1.6.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6028, commit ID: 084e4e126211d74a79e8dbd2d0e604dd3c650822

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 22]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createWithDefault(64)
 
   private[spark] val RPC_IO_NUM_CONNECTIONS_PER_PEER =
     ConfigBuilder("spark.rpc.io.numConnectionsPerPeer")
+      .version("1.6.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-10745, commit ID: 34a77679877bc40b58a10ec539a8da00fed7db39

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 23]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createWithDefault(1)
 
   private[spark] val RPC_IO_THREADS =
     ConfigBuilder("spark.rpc.io.threads")
+      .version("1.6.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6028, commit ID: 084e4e126211d74a79e8dbd2d0e604dd3c650822

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 24]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createOptional
 
   private[spark] val RPC_LOOKUP_TIMEOUT =
     ConfigBuilder("spark.rpc.lookupTimeout")
+      .version("1.4.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 25]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .stringConf
       .createOptional
 
   private[spark] val RPC_MESSAGE_MAX_SIZE =
     ConfigBuilder("spark.rpc.message.maxSize")
+      .version("2.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-7997, commit ID: bc1babd63da4ee56e6d371eb24805a5d714e8295

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 26]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createWithDefault(128)
 
   private[spark] val RPC_NETTY_DISPATCHER_NUM_THREADS =
     ConfigBuilder("spark.rpc.netty.dispatcher.numThreads")
+      .version("1.6.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-11079, commit ID: 1797055dbf1d2fd7714d7c65c8d2efde2f15efc1

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 27]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createOptional
 
   private[spark] val RPC_NUM_RETRIES =
     ConfigBuilder("spark.rpc.numRetries")
+      .version("1.4.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae

File: core/src/main/scala/org/apache/spark/internal/config/Network.scala [hunk 28]
```diff
@@ -23,71 +23,85 @@ private[spark] object Network {
...
       .intConf
       .createWithDefault(3)
 
   private[spark] val RPC_RETRY_WAIT =
     ConfigBuilder("spark.rpc.retry.wait")
+      .version("1.4.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Network.scala]: SPARK-6490, commit ID: 8136810dfad12008ac300116df7bc8448740f1ae

**Comment by beliefer** [comment]
retest this please

File: docs/configuration.md [hunk 29]
```diff
@@ -1758,7 +1766,8 @@ Apart from these, the following properties are also available, and may be useful
     Off-heap buffers are used to reduce garbage collection during shuffle and cache
     block transfer. For environments where off-heap memory is tightly limited, users may wish to
     turn this off to force all allocations to be on-heap.
-    </td>
+  </td>
+  <td></td>
```
Comment (HyukjinKwon) [inline docs/configuration.md]: @beliefer, can we add the version by tracking `common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java`?

**Review by HyukjinKwon — APPROVED** [review]
Looks fine. I will merge after tacking another look if this doesn't get merged in few days.

**Comment by beliefer** [comment]
> Looks fine. I will merge after tacking another look if this doesn't get merged in few days.

Thanks!

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by beliefer** [comment]
@HyukjinKwon Thanks for your help!

**Comment by HyukjinKwon** [comment]
Merged to branch-3.0 too

**Pull request closed without merging**